enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -147,6 +149,26 @@ func (en *executionNode) GetExecutorID() string {
 	return en.executorID
 }
 
+func toNodeInterfaces(nodes []*executionNode) []interfaces.ExecutionNode {
+	out := make([]interfaces.ExecutionNode, 0, len(nodes))
+	for _, node := range nodes {
+		out = append(out, node)
+	}
+	return out
+}
+
+func fromNodeInterfaces(nodes []interfaces.ExecutionNode) ([]*executionNode, error) {
+	out := make([]*executionNode, 0, len(nodes))
+	for _, node := range nodes {
+		en, ok := node.(*executionNode)
+		if !ok {
+			return nil, status.InternalError("failed to convert executionNode to interface; this should never happen")
+		}
+		out = append(out, en)
+	}
+	return out, nil
+}
+
```

Comment:
Why are these necessary / is there a better way?

Reply:
Thanks for the explanation!

That interface change seems like a great way to clean this up, do you plan to do that in a followup?

Reply:
Yep, I will work on that after merging this one.